src/generate_pisa_index.cpp

```diff
@@ -21,19 +41,138 @@ std::ostream &write_sequence(std::ostream &os, gsl::span<T> sequence)
 }
 
 
+struct inverted_index {
+
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_documents {};
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_frequencies {};
+  std::map<uint32_t, uint32_t> m_document_sizes {};
+  std::vector<std::string> m_plain_terms {};
+ 
+  void add_document_lengths(const std::string doclen_file) {
+      std::ifstream input_lengths(doclen_file);
+      uint32_t doc_id;
+      uint32_t size;
+      while (input_lengths >> doc_id >> size) {
+          m_document_sizes[doc_id] = size;
+      }  
+      std::cerr << "Read " << m_document_sizes.size() << " document lengths."
+                << std::endl;
+  } 
+
+  void add_postings_list(const PostingsList& postings_list, uint32_t term_id) {
+
+      if (term_id % 10000 == 0) {
+          std::cerr << "Processing list " << term_id << "..." << std::endl;
+      }
+
+      std::string term = postings_list.term();
+      m_plain_terms.emplace_back(term);
+
+      uint32_t doc_freq = postings_list.df();
+      // uint32_t collection_freq = postings_list.cf();
```

Comment:
Did you mean to delete it or is it supposed to be used? Btw. what's document frequency and what is collection frequency? Is the first number of postings, and the second number of occurrences (sum of freqs in the list)?

Reply:
Don't need `collection_freq` - I will delete it.
You are correct - `doc_freq` is the no. docs containing the term, and `collection_freq` is the sum of the freqs in the list.